### PR TITLE
refactor: text field guideline update

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes

The primary change to look for is the variation of the font colors based on which `fieldType` the input is. The `tertiary` is a new addition that is used already in the custom links so there are no pre-existing implementations of it.

The `compact` property is now gone and was replaced by being the `secondary` field type.

Primary is the common and default implementation of the text field.

This now supports icons inside the text field itself and the padding varies between having one or not.

## Testing

- Did you introduce testing for this PR
- ☑️ Did you test the webapp? **Yes**
- ☑️ Did you test the extension? **Yes**

DD-495 #done

**Note:** just waiting for Tsahi's answer regarding the right icon but this should be alright to be reviewed.

### CODE REVIEW

File: packages/shared/src/components/fields/TextField.spec.tsx [hunk 1]
```diff
@@ -58,7 +58,7 @@ it('should set hint role as alert when invalid', async () => {
 });
 
 it('should show both label and placeholder in compact mode', async () => {
```
Comment (rebelchris) [inline packages/shared/src/components/fields/TextField.spec.tsx:60]: Can we rename this to `secondary` mode? (The actual test)

Comment (rebelchris) [inline packages/shared/src/components/fields/TextField.spec.tsx:60]: Also can we add a specific test for this new one?

File: packages/shared/src/components/fields/TextField.tsx [hunk 2]
```diff
@@ -52,6 +56,9 @@ export function TextField({
     onInput: baseOnInput,
     focusInput,
   } = useInputField(value, valueChanged);
+  const isPrimary = fieldType === 'primary';
+  const isSecondary = fieldType === 'secondary';
+  const isTertiary = fieldType === 'tertiary';
```
Comment (rebelchris) [inline packages/shared/src/components/fields/TextField.tsx:61]: Could we maybe make this a bit more explicit?

**Review by rebelchris — CHANGES_REQUESTED** [review]
Very nice setup!
Would only state we could add some more tests around the specific types?

Perhaps we can add to check for certain classes?
This will quickly show us if we ever move one by accident

What do you think?

**Comment by sshanzel** [comment]
@rebelchris thank you for the review! Totally agree with it! This is a reused component across the app, tests must be very strict in this scenario.

**Review by rebelchris — APPROVED** [review]
Nice work on this

**Review by idoshamun — COMMENTED** [review]
Awesome!

**Pull request merged**